pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/dao/AnswerDao.java

```diff
@@ -154,11 +154,43 @@ private void createAnswerFileValue(long answerId, FileAnswer answer) {
         if (answer.getValue() == null) {
             return;
         }
-        List<Long> uploadIds = answer.getValue().stream().map(FileInfo::getUploadId).collect(Collectors.toList());
-        int[] inserted = getAnswerSql().bulkInsertFileValue(answerId, uploadIds);
+
+        final List<Long> uploadIds = answer.getValue().stream().map(FileInfo::getUploadId).collect(Collectors.toList());
+        final int[] inserted = getAnswerSql().bulkInsertFileValue(answerId, uploadIds);
         if (inserted.length != uploadIds.size()) {
-            throw new DaoException("Not all file uploads were assigned to answer " + answerId);
+            final var message = String.format("Failed to insert uploads ids for [answer:%s] (%s inserted, %s expected)",
+                    answerId,
+                    inserted.length,
+                    uploadIds.size());
+            throw new DaoException(message);
+        }
+
+        if (uploadIds.isEmpty()) {
+            // No file uploads specified- nothing more to do here.
+            return;
         }
+
+        /*
+         * The answer dto needs to be pulled directly here as the `Answer` class does not
+         *  include the lastUpdatedAt time. It could be added- the ideal spot would likely be
+         *  in {@link AnswerWithValueReducer}- but that would involve altering the ctors for a
+         *  very widely used base class. This should work for the time being.
+         *  (bskinner - 20221007)
+         */
+        final var uploadedAt = getAnswerSql()
+                .findDtoById(answerId)
+                .map(AnswerDto::getLastUpdatedAt)
+                .map(timestamp -> Instant.ofEpochMilli((long)timestamp))
```

Comment:
Could be replaced with
```
  .mapToLong(Long::longValue)
  .map(Instant::ofEpochMilli)
```

Reply:
Not in the current form- `mapToLong()` is a method on the `Stream` class, not on the `Optional<T>` class.